packages/starlight/user-components/LinkCard.astro

```diff
@@ -1,20 +1,36 @@
 ---
 import Icon from './Icon.astro';
+import Badge from '../components/Badge.astro';
+import type { Badge as BadgeType } from '../schemas/badge';
+import { BadgeConfigSchema } from '../schemas/badge';
 import type { HTMLAttributes } from 'astro/types';
 
 interface Props extends Omit<HTMLAttributes<'a'>, 'title'> {
 	title: string;
+	badge: BadgeType | undefined;
 	description?: string;
 }
 
-const { title, description, ...attributes } = Astro.props;
+const { title, badge, description, ...attributes } = Astro.props;
+
+const badgeData = BadgeConfigSchema().parse(badge);
```

Comment:
```suggestion
const badgeData = BadgeConfigSchema().parse(badge, { errorMap });
```

If an invalid prop is passed down, without using the Starlight error map, the user will see:

<img width="308" alt="SCR-20230912-qylk" src="https://github.com/withastro/starlight/assets/494699/0171ff46-3332-4ead-8919-8fc7d77e0827">

When using the error map, the error becomes:

<img width="603" alt="SCR-20230912-qypb" src="https://github.com/withastro/starlight/assets/494699/96c548ef-105f-4a74-94be-aed4a8360746">

Altho, I wonder if we should use `parse` here and not `safeParse` and re-throw a better error considering that with this setup, the error would be reported in `lib/index.mjs:537:31` or something which not super helpful.

Any thoughts? (an example on how to use `safeParse` is available in `packages/starlight/index.ts`)

Reply:
Awesome! Thanks for the suggestion! If we use `safeParse` then that would enable better error surfacing if I'm understanding your comment correctly, right? If so then I'd get behind that.

Reply:
Yeah, we could do something like this technically:

```ts
import { AstroError } from 'astro/errors';

// …

const badgeData = BadgeConfigSchema().safeParse(badge, { errorMap });

if (!badgeData.success) {
	throw new AstroError(
		'Invalid `badge` prop passed down to `<LinkCard>`.',
		'Find out more in our documentation at https://starlight.astro.build/reference/frontmatter/#sidebar'
	);
}

// …

// Here, we know the data is valid.
badgeData.data;
```

Which would give us maybe a better error:

<img width="1052" alt="image" src="https://github.com/withastro/starlight/assets/494699/65e33c52-3fea-4c4f-b3ba-02283a4e4c9e">

Is this any better? I think this would be the first time we would use zod in a user component, so not sure what would be the best way to surface this kind of error.

Reply:
Thanks! I tweaked the error message slightly from your suggestion to hopefully make it a little more clear:

<img width="1218" alt="Screenshot 2023-09-12 at 14 00 00" src="https://github.com/withastro/starlight/assets/15347255/b78e305a-2b3a-4d15-8857-68c0ea1a1bd7">